Learning/repeat

Paullina26 opened the pull request.

Paullina26 committed:
settings language correction

Paullina26 committed:
Errors StyleComponents missing $ -> positionTop

positionTop to są style, więc jak nie dasz $ przed nazwą to będą błędy w consoli.

Paullina26 committed:
Removed warning about passing styles to the dom element

Paullina26 committed:
add endpoint lernedWords

Paullina26 committed:
Preliminary version repeat

- Weekly vocabulary support
- new style of word display components

Paullina26 committed:
correction

zoska91 commented on src/API/api.ts:
```diff
@@ -1,37 +1,40 @@
-// export const API_BASE_URL = 'http://localhost:8080/api/';
+// Podstawowe URL i nagłówki
 export const API_BASE_URL = 'https://onewordserv.byst.re/api/';
```
komentarze po angielsku

zoska91 commented on src/components/Shared/Atoms/Title.tsx:
```diff
@@ -5,6 +5,7 @@ export const TitleBig = styled.h2`
   text-align: center;
   ${font_settings(3, 'ita', 500)};
   color: ${({ theme }) => theme.purpleDark};
+  margin: 5px auto;
```
takich styli lepiej nie dodawać do Atomu, bo później będziesz chciałą gdzieś to jeszcze wykorzystać i ten margin może przeszkadzać. Lepiej w docelowym miejscu zrobić coś w stylu: 

```
const StyledTitle = styled(Title)`
  margin: 5px auto;
`
```

Paullina26 replied:
Z założenia ma być to tytuł, który zawsze jest  w jakimś kontenerze i powinien mieć identyczne zachowanie. 
Używamy tego komponentu w kilku miejscach i ma dokładnie identyczny styl wszędzie. 
Żeby to zmienić, trzeba w kilku miejscach teraz napisać taki sam kawałek kodu: 

const StyledTitle = styled(Title)`
  margin: 5px auto;
`

zoska91 replied:
ok, skoro tak to może zostać

zoska91 commented on src/index.tsx:
```diff
@@ -2,6 +2,7 @@ import ReactDOM from 'react-dom/client';
 import App from './App';
 
+
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
```
lepiej nie zostrawiać 2 linii przerwy (?)

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+        method: 'GET',
+        queryParams: { days: 7, limit: 100 },
+      });
+      console.log('Response', result);
```
console.log wywal

Paullina26 replied:
Komponent jest jeszcze do poprawek po testach. 
Te console.log pomagają sprawdzać jakie dane przychodzą z backendu do czasu ukończenia tego komponentu lepiej je zostawić, by sprawdzać przepływ danych.

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+      if (result.response.words.length > 0) {
+        setWordBase(result.response.words[0].basicWord);
+      }
+    } catch (error) {
```
czemu żeby dodać do wordBase sprawdzasz czy są jakieś słowa, a dla wordsReapeat nie?

Paullina26 replied:
Dostawałam jakiś dziwny błąd na etapie pisania kodu od TS stąd robiłam to sprawdzanie by zgadzały się dane.  
Może dodać do obydwu i gdy długość będzie zero wyświetlić stosowną informację.

zoska91 replied:
no lepiej do obu dodać sprawdzenie. a pamiętasz jaki błąd?

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+  }, []);
+
+  useEffect(() => {
+    console.log('Words repeat updated:', wordsRepeat);
```
:(

Paullina26 replied:
jak wyżej w obecnej fasie testowania i wprowadzania zmian, te logi pomagają śledzić przepływ danych.

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+  return (
+    <S.Wrapper>
+      <TitleBig>Repeat</TitleBig>
+      <S.WrapperBaseWord>{wordBase ? wordBase : 'No Words'}</S.WrapperBaseWord>
```
`{wordBase ? wordBase : 'No Words'}` taki zapis możesz zastąpić: 
`wordBase || 'No words'`

On wtedy wyświetli wordBase, a jak nie bęzie wordBase to 'No words'

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+    <S.Wrapper>
+      <TitleBig>Repeat</TitleBig>
+      <S.WrapperBaseWord>{wordBase ? wordBase : 'No Words'}</S.WrapperBaseWord>
+      <div>{feedback && <div>{feedback}</div>}</div>
```
po co ten div w środku? Jeśli ma coś tam podkreślać, mieć jakiś inny styl to lepiej wykorzystać `span`

Paullina26 replied:
Jeszcze brak pomysłu na to jak miałoby to wyglądać. 
Prawdopodobnie będzie tam osobny komponent do wyświetlania informacji o wynikach.

zoska91 commented on src/pages/User/Learning/Reapeat/Repeat.tsx:
```diff
@@ -0,0 +1,105 @@
+        />
+      </div>
+      <S.WrapperButton>
+        <ButtonIcon nameIcon='back' $margin='2px auto' onClick={handleBackWord} />
```
dodaj tutaj warunki, żeby przycisk `back` byl niewidoczny albo wyszarzony (żeby się układ nie posypał) gdy aktualne słowo jest pierwszym. 
I tak samo z ostatnio. 

Chyba, że to w jakiejś pędli będzie (?)

zoska91 replied:
ale myślę, że lepiej nie robić pętli tylko jak ktoś zgadnie wszystkie to jeszcze dać coś z tylu, że brawo wszystko dobrze! Albo wręcz zrobić w formie quizu i na koniec dać userowi wynik np: 3/4

Paullina26 replied:
dodaje jako osobne zadanie.

Paullina26 committed:
CR

Paullina26 committed:
Added button activation and deactivation functionality

Added functionality to activate and deactivate the Button component. This allows managing the button state without altering the page layout, providing clear communication to users when the button is inactive

Paullina26 committed:
Deactivating buttons

Buttons dynamically change state.
In certain cases, the button is disabled and the style is changed.

Paullina26 committed:
RepeatWords, Form Repeat

addition of the RepeatWords component, a new look for the Repeat website.
Initial setting of the Repeat form.

Paullina26 committed:
add form repeat

zoska91 commented on src/components/RepeatWords/RepeatWords.tsx:
```diff
@@ -0,0 +1,137 @@
+  updatedDate: string;
+  daysRepeat: number;
+  __v: number;
+}
```
może typ warto doi osobnego pliku wyciągnać, bo plik sie juz troche wydłużył...

zoska91 commented on src/components/RepeatWords/RepeatWords.tsx:
```diff
@@ -0,0 +1,137 @@
+
+interface RepeatWordsProps {
+  daysRepeat: number;
+}
```
i ten tez tak mozesz dorzucić.

zoska91 commented on src/components/RepeatWords/RepeatWords.tsx:
```diff
@@ -0,0 +1,137 @@
+
+const RepeatWords: React.FC<RepeatWordsProps> = ({ daysRepeat }) => {
+  const [wordsRepeat, setWordsRepeat] = useState<Word[]>([]);
+  const [currentWordIndex, setCurrentWordIndex] = useState<number>(0);
```
tu nie musisz robic `<number>`, bo jeśli value jest number to ts sam o tym wie. 
wyżej jest to ptrzebne bo samo `[]` nic mus nie mówi
tam samo niżej `<string>` nie jest potrzebne. on wie, ze jak jest `''` to będzie typ string

zoska91 commented on src/API/api.ts:
```diff
@@ -40,12 +43,20 @@ type DynamicEndpoint = {
 
+// Interface for query parameters
+interface QueryParams {
+  limit?: number;
```
w całej aplikacji to jedyne paramsy jakie mamy? 
limit idays? żeby mogłaś to tak zostawić? trochę to jest dziwne. przecież każdy endpoint może mieć swoje paramsy, więc tu by się lepiej jakiś generk pewnie sprawdził. 
na razie zostaw, ale chcę Ci zwrócić uwagę na to, że to bardzo podejrzane :)

zoska91 commented on src/components/Repeat/RepeatForm/RepeatForm.tsx:
```diff
@@ -0,0 +1,42 @@
+
+const RepeatForm: FC<RepeatFormProps> = ({ daysRepeat, onDaysRepeatChange, onSubmit }) => {
+  const handleDaysRepeatChange = (value: string) => {
+    const numericValue = parseInt(value, 10);
```
`const numericValue = parseInt(value, 10);`
zamiast tego możesz też uzyć `+value`. Ale to juz jak chcesz. bez różnicy :)

zoska91 commented on src/components/Repeat/RepeatWords/RepeatWords.tsx:
```diff
@@ -0,0 +1,137 @@
+interface RepeatWordsProps {
+  daysRepeat: number;
+}
+
```
te typu warto wyciagnać do osobnego pliku jak Ci się tak ten rozrósł

zoska91 commented on src/pages/User/Learning/Repeat/Repeat.tsx:
```diff
@@ -0,0 +1,41 @@
+
+const Repeat: React.FC<RepeatProps> = () => {
+  const [daysRepeat, setDaysRepeat] = useState<number>(7);
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
```
jeśli typ jes tak oczywisty jak tutaj to nie musisz go deklarować. gdy value = `7` to ts wie, ze daysRepeat będzie `number`
i tak samo jak jest `false` to wie, ze będzie `boolean`

zoska91 commented on src/pages/User/Learning/Repeat/Repeat.tsx:
```diff
@@ -0,0 +1,41 @@
+
+  const handleSubmit = () => {
+    if (daysRepeat >= 1 && daysRepeat <= 30) {
+      console.log('Days Repeat:', daysRepeat);
```
:(

Paullina26 committed:
CR
